crates/uv/tests/common/mod.rs

```diff
@@ -168,10 +172,35 @@ impl TestContext {
     ///
     /// See [`TestContext::new`] if only a single version is desired.
     pub fn new_with_versions(python_versions: &[&str]) -> Self {
-        let temp_dir = assert_fs::TempDir::new().expect("Failed to create test working directory");
-        let cache_dir = assert_fs::TempDir::new().expect("Failed to create test cache directory");
-        let python_dir = assert_fs::TempDir::new().expect("Failed to create test Python directory");
-        let home_dir = assert_fs::TempDir::new().expect("Failed to create test home directory");
+        // Discover the path to the XDG state directory. We use this, rather than the OS-specific
+        // temporary directory, because on macOS (and Windows on GitHub Actions), they involve
+        // symlinks. (On macOS, the temporary directory is, like `/var/...`, which resolves to
+        // `/private/var/...`.)
+        //
+        // It turns out that, at least on macOS, if we pass a symlink as `current_dir`, it gets
+        // _immediately_ resolved (so that if call `current_dir` in the subsequent command, you get
+        // the resolved symlink). This is problematic, as we _don't_ want to resolve symlinks for
+        // user paths.
+        let bucket = etcetera::base_strategy::choose_base_strategy()
+            .expect("Failed to find base strategy")
+            .data_dir()
+            .join("uv")
+            .join("tests");
+        fs_err::create_dir_all(&bucket).expect("Failed to create test bucket");
+
+        let root = tempfile::TempDir::new_in(bucket).expect("Failed to create test root directory");
+
+        let temp_dir = ChildPath::new(root.path()).child("temp");
+        fs_err::create_dir_all(&temp_dir).expect("Failed to create test working directory");
+
+        let cache_dir = ChildPath::new(root.path()).child("cache");
+        fs_err::create_dir_all(&cache_dir).expect("Failed to create test cache directory");
+
+        let python_dir = ChildPath::new(root.path()).child("python");
+        fs_err::create_dir_all(&python_dir).expect("Failed to create test Python directory");
+
+        let home_dir = ChildPath::new(root.path()).child("home");
+        fs_err::create_dir_all(&home_dir).expect("Failed to create test home directory");
 
         // Canonicalize the temp dir for consistent snapshot behavior
         let canonical_temp_dir = temp_dir.canonicalize().unwrap();
```

Comment:
Can probably drop this hack too?

Reply:
I think this needs to come in a future PR, since we still canonicalize internally...

Reply:
In theory canonical temporary directory for testing == non-canonical now, right?